animations-dbus/animations-dbus-server-object.c

```diff
@@ -735,3 +744,68 @@ animations_dbus_server_new_with_connection_async (AnimationsDbusServerEffectFact
                               "effect-factory", factory,
                               NULL);
 }
+
+/**
+ * animations_dbus_server_stop:
+ * @self: an #AnimationsDbusServer
+ * @cancellable: (nullable): a #GCancellable, or %NULL
+ * @error: return location for a #GError, or %NULL
+ *
+ * Stop the server, unexport all D-Bus objects, and unown the well-known
+ * D-Bus name for the service.
+ *
+ * Returns: %TRUE on success, %FALSE otherwise
+ */
+gboolean
+animations_dbus_server_stop (AnimationsDbusServer  *self,
+                             GCancellable          *cancellable,
+                             GError               **error)
```

Comment:
Can we restart the server after calling `stop()`? If not, probably worth a note in the docs. Alternatively we could make this private and only invoked from dispose - for the hack extension we could force a call to `animationDbusServer.run_dispose()` when stopping it.

Reply:
No, it can’t be restarted. I’ve added that to the documentation. Forcing a call to `run_dispose()` is almost always a hack (because it interferes with the strong refs still potentially held by other code) so I’m not going to do that. I think it’s fine for the object to essentially be useless (but harmless) after `stop()` is called.